pkg/detector/detector.go

```diff
@@ -131,36 +132,41 @@ func (d *ResourceDetector) Start(ctx context.Context) error {
 	d.clusterPolicyReconcileWorker.Run(d.ConcurrentClusterPropagationPolicySyncs, d.stopCh)
 
-	// watch and enqueue PropagationPolicy changes.
-	propagationPolicyGVR := schema.GroupVersionResource{
-		Group:    policyv1alpha1.GroupVersion.Group,
-		Version:  policyv1alpha1.GroupVersion.Version,
-		Resource: policyv1alpha1.ResourcePluralPropagationPolicy,
-	}
-	policyHandler := fedinformer.NewHandlerOnEvents(d.OnPropagationPolicyAdd, d.OnPropagationPolicyUpdate, nil)
-	d.InformerManager.ForResource(propagationPolicyGVR, policyHandler)
-	d.propagationPolicyLister = d.InformerManager.Lister(propagationPolicyGVR)
-
-	// watch and enqueue ClusterPropagationPolicy changes.
-	clusterPropagationPolicyGVR := schema.GroupVersionResource{
-		Group:    policyv1alpha1.GroupVersion.Group,
-		Version:  policyv1alpha1.GroupVersion.Version,
-		Resource: policyv1alpha1.ResourcePluralClusterPropagationPolicy,
-	}
-	clusterPolicyHandler := fedinformer.NewHandlerOnEvents(d.OnClusterPropagationPolicyAdd, d.OnClusterPropagationPolicyUpdate, nil)
-	d.InformerManager.ForResource(clusterPropagationPolicyGVR, clusterPolicyHandler)
-	d.clusterPropagationPolicyLister = d.InformerManager.Lister(clusterPropagationPolicyGVR)
-
 	detectorWorkerOptions := util.Options{
 		Name:               "resource detector",
 		KeyFunc:            ResourceItemKeyFunc,
 		ReconcileFunc:      d.Reconcile,
 		RateLimiterOptions: d.RateLimiterOptions,
 	}
-
-	d.EventHandler = fedinformer.NewFilteringHandlerOnAllEvents(d.EventFilter, d.OnAdd, d.OnUpdate, d.OnDelete)
 	d.Processor = util.NewAsyncWorker(detectorWorkerOptions)
 	d.Processor.Run(d.ConcurrentResourceTemplateSyncs, d.stopCh)
+
+	// watch and enqueue PropagationPolicy changes.
+	policyHandler := fedinformer.NewHandlerOnEvents(d.OnPropagationPolicyAdd, d.OnPropagationPolicyUpdate, nil)
+	ppInformer, err := d.ControllerRuntimeCache.GetInformer(ctx, &policyv1alpha1.PropagationPolicy{})
+	if err != nil {
+		klog.Errorf("Failed to get informer for PropagationPolicy: %v", err)
+		return err
+	}
+	_, err = ppInformer.AddEventHandler(policyHandler)
+	if err != nil {
+		klog.Errorf("Failed to add event handler for PropagationPolicy: %v", err)
+		return err
+	}
+
+	// watch and enqueue ClusterPropagationPolicy changes.
+	clusterPolicyHandler := fedinformer.NewHandlerOnEvents(d.OnClusterPropagationPolicyAdd, d.OnClusterPropagationPolicyUpdate, nil)
+	cppInformer, err := d.ControllerRuntimeCache.GetInformer(ctx, &policyv1alpha1.ClusterPropagationPolicy{})
+	if err != nil {
+		klog.Errorf("Failed to get informer for ClusterPropagationPolicy: %v", err)
+		return err
+	}
+	_, err = cppInformer.AddEventHandler(clusterPolicyHandler)
+	if err != nil {
+		klog.Errorf("Failed to add event handler for ClusterPropagationPolicy: %v", err)
+		return err
+	}
+	d.EventHandler = fedinformer.NewFilteringHandlerOnAllEvents(d.EventFilter, d.OnAdd, d.OnUpdate, d.OnDelete)
 	go d.discoverResources(30 * time.Second)
```

Comment:
I would suggest adding a empty line before line 169, or moving this part after `d.Processor.Run()`

Reply:
d.EventHandler  is used in function `d. discoverResources `, I mean create d. EventHandler before  `d.discoverResources` is enough maybe

Reply:
What I mean is that `Processor`, `ppHandler`, `cppHandler`, each of them has a blank line as a separator. The two lines I quoted (169, 170) have little connection with cpp, but there is no separation between them and cppHandler, and they look like a whole. So I suggest either adding a blank line to separate them, or putting these two lines together with the Processor part above because they are more closely related.

Reply:
I got it